### PR TITLE
Plfm 329 logger interface

### PR DESCRIPTION
**Pull request by mykelswitzer**
these are fixes that make it work basically... tested this by using the logging interface in the identity service, and it worked... though it did require me still updating all logging import paths to v2... but the interface only exists in v2....

this should just be a stopgap "fix and then we probably should move these cross-cutting packages into individual repos... like

gopgk-logging or something that makes it clear they're all go packages

### CODE REVIEW

**Review by shyhshyanhsu — APPROVED** [review]
Looks good to me.  Please have more reviewers to review.

File: pkg/logging/field.go [hunk 1]
```diff
@@ -2,11 +2,19 @@ package logging
 
 import "go.uber.org/zap"
 
+type DataField interface {
+	GetField() zap.Field
```
Comment (amarine7882) [inline pkg/logging/field.go:6]: Would this be better as an internal method?

Comment (amarine7882) [reply pkg/logging/field.go:6]: Sorry, meant like this
```golang
type DataField interface {
	getField() zap.Field
}
```
I think private interface methods are still compatible across versions, and this prevents unnecessary exposure of the zap abstraction.

Comment (mykelswitzer) [reply pkg/logging/field.go]: apparently you can

**Review by amarine7882 — APPROVED** [review]

**Pull request merged**